lib/preprocessor.js

```diff
@@ -60,6 +62,15 @@ var createCoveragePreprocessor = function(logger, basePath, reporters, coverageR
       // remember the actual immediate instrumented JS for given original path
       sourceCache[jsPath] = content;
 
+      if (includeAllSources) {
```

Comment:
Once we have that instrumented version of the file you mentioned, it contains in it an object that it uses to track when lines, branches, etc are called. This object is normally exposed to the reporters as `result.coverage` (see https://github.com/mwinche/karma-coverage/blob/master/lib/reporter.js#L126).

What I am doing is pulling out the initialized version of that `.coverage` object using the regex below. I borrowed that regex from gulp-istanbul. Once I have the initialized version of coverage object, I store it in the `stubs` object for later retrieval during the reporting phase initialization. If you have suggestions on better ways to retrieve this coverage object, I'd love to hear it!

The `stubs` object collects initialized coverage objects for all preprocessed files and provides a method for retrieving them all at once (`get`) in the format that Istanbul Collectors desire.=`

In the start of the reporting phase then we just need add these initialized coverage objects (I'm calling them stubs) to the collector of each browser. As normal test execution proceeds we will add actual result coverage data which will override the data in the stubs.

Effectively this gives a default for all preprocessed files of zero coverage that will be updated as tests run.

To be fair, `stubs` was the best name I could think of, if you have better idea I'm all ears!